### ISSUE TITLE
buglet in basic http auth

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10361]
|<img alt="mgedmin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| mgedmin reported|
|-|-|
|Trac ID|trac#479|
|Type|defect|
|Created|2004-01-16 17:30:00Z|

<details><summary>Searchable metadata</summary>

```
trac-id__479 479
type__defect defect
reporter__mgedmin mgedmin
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1074274200000000 1074274200000000
changetime__1470903604121729 1470903604121729
version__ 
owner__itamarst itamarst
cc__itamarst cc__mgedmin
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="mgedmin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|mgedmin commented|
|-|-|

```
#!html
<pre>
There are one or two minor problems with the _authorize method of
twisted.protocols.http.Request:

  1) according to RFC 2617 the password may contain a colon, therefore
            self.user, self.password = upw.split(':')
     should be replaced with
            self.user, self.password = upw.split(':', 1)

  2) the base64-encoded user:password pair is only defined for basic HTTP
     auth, so perhaps self.user and self.password should only be set if
     bas.lower() == "basic".
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Fixed in revision 10305.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub commented|
|-|-|

In changeset 343f4c0f0dc27c8129a2481c0ccb7a7487a58c6f
```
#!CommitTicketReference repository="" revision="343f4c0f0dc27c8129a2481c0ccb7a7487a58c6f"
Merge pull request #479 from twisted/8735-rodrigc-process-py3

Author: hawkowl, rodrigc
Reviewer: glyph
Fixes: #8735

Fix process related tests on Windows Python 3
```
